system-contract-dapp-playground/src/components/contract-interaction/hts/token-create-custom/shared/sharedComponents.tsx

```diff
@@ -0,0 +1,793 @@
+/*-
+ *
+ * Hedera Smart Contracts
+ *
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+import { ethers } from 'ethers';
```

Comment:
Erm this file is pretty big and seems to have lots of components that could maybe be pulled out into their own component files.
Is there a reason to keep them all in one file?